src/HybridAppSignIn.tsx

```diff
@@ -0,0 +1,57 @@
+import {findFocusedRoute} from '@react-navigation/native';
+import {useContext, useEffect, useState} from 'react';
+import {Linking} from 'react-native';
+import type {AppProps} from './App';
+import CONST from './CONST';
+import {signInAfterTransitionFromOldDot} from './libs/actions/Session';
+import Navigation, {navigationRef} from './libs/Navigation/Navigation';
+import type {Route} from './ROUTES';
+import ROUTES from './ROUTES';
+import SCREENS from './SCREENS';
+import SplashScreenStateContext from './SplashScreenStateContext';
+
+function handleHybridUrlNavigation(url: Route) {
+    const parsedUrl = Navigation.parseHybridAppUrl(url);
+
+    Navigation.isNavigationReady().then(() => {
+        if (parsedUrl.startsWith(`/${ROUTES.SHARE_ROOT}`)) {
+            const focusRoute = findFocusedRoute(navigationRef.getRootState());
+            if (focusRoute?.name === SCREENS.SHARE.SHARE_DETAILS || focusRoute?.name === SCREENS.SHARE.SUBMIT_DETAILS) {
+                Navigation.goBack(ROUTES.SHARE_ROOT);
+                return;
+            }
+        }
+        Navigation.navigate(parsedUrl);
+    });
+}
+
+function HybridAppSignIn({url, hybridAppSettings}: AppProps) {
+    const [signInHandled, setSignInHandled] = useState(false);
+    const {setSplashScreenState} = useContext(SplashScreenStateContext);
+
+    useEffect(() => {
+        const listener = Linking.addEventListener('url', (state) => {
+            handleHybridUrlNavigation(state.url as Route);
+        });
+
+        return () => {
+            listener.remove();
+        };
+    }, []);
+
+    if (!url || !hybridAppSettings || signInHandled) {
+        return null;
+    }
+
+    signInAfterTransitionFromOldDot(hybridAppSettings).then(() => {
```

Comment:
Hmmm this could be kept in `useEffect` as well

Reply:
I'm not so sure because I remember that sometimes `url` so as `hybridAppSettings` were undefined (early in the app initialization) and it's safer to make this code rerunable